File: src/AutoContext.Worker.Shared/Hosting/WorkerHostOptions.cs
namespace AutoContext.Worker.Hosting;

/// <summary>
/// Strongly-typed options for the shared worker host: the named pipe to
/// listen on and the stderr ready-marker emitted once the pipe is accepting
/// connections.
/// </summary>
/// <remarks>
/// Each <c>AutoContext.Worker.*</c> process binds and registers an instance
/// of this type. <see cref="Pipe"/> normally comes from the <c>--pipe</c>
/// command-line argument. <see cref="ReadyMarker"/> is supplied by the
/// process itself (e.g. <c>"[AutoContext.Worker.DotNet] Ready."</c>) and is
/// scraped from the worker's stderr by the parent process to detect that
/// the pipe is up.
/// </remarks>
public sealed class WorkerHostOptions
{
    /// <summary>
    /// Named pipe the worker listens on for per-task requests.
    /// </summary>
    public string Pipe { get; init; } = string.Empty;

    /// <summary>
    /// Exact text written to stderr once the pipe server is accepting
    /// connections. Used by parent processes as a readiness handshake.
    /// </summary>
    public string ReadyMarker { get; init; } = string.Empty;
}

File: src/AutoContext.Worker.Web/src/hosting/worker-host-options.ts
/**
 * Strongly-typed options for a worker host: the named pipe to listen on
 * and the stderr ready-marker emitted once the pipe is accepting
 * connections.
 *
 * Each `AutoContext.Worker.*` process supplies an instance. `pipe`
 * normally comes from the `--pipe` command-line argument.
 * `readyMarker` is process-specific (e.g. `[AutoContext.Worker.Web] Ready.`)
 * and is scraped from the worker's stderr by the parent process to detect
 * that the pipe is up.
 */
export interface WorkerHostOptions {
    /**
     * Named pipe the worker listens on for per-task requests. May be a
     * bare name (`autocontext.web-worker`) or a fully-qualified path;
     * the service normalizes bare names to `\\.\pipe\<name>` on Windows.
     */
    readonly pipe: string;

    /**
     * Exact text written to stderr once the pipe server is accepting
     * connections. Used by parent processes as a readiness handshake.
     */
    readonly readyMarker: string;
}
